{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# sudo -H anaconda/bin/pip install biopython\n",
    "\n",
    "from Bio import Entrez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Entrez.email = 'greg.clinton@gmail.com'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# How many genes does e. coli have? What are their names?\n",
    "# How many species under homo?\n",
    "# sra vs nucleotide, gene, ref_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "record = Entrez.read(Entrez.esearch(db = 'gene', term = 'e coli[Orgn]'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'947170,948466,947777,948511,947071'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = ','.join(record['IdList'][:5])\n",
    "ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "records = Entrez.read(Entrez.efetch(db = 'gene', id = ids, retmode = 'xml'))\n",
    "len(records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[u'Entrezgene_rna', u'Entrezgene_unique-keys', u'Entrezgene_track-info', u'Entrezgene_comments', u'Entrezgene_locus', u'Entrezgene_source', u'Entrezgene_type', u'Entrezgene_gene-source', u'Entrezgene_xtra-iq', u'Entrezgene_non-unique-keys', u'Entrezgene_gene', u'Entrezgene_properties']\n"
     ]
    }
   ],
   "source": [
    "print records[0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n"
     ]
    }
   ],
   "source": [
    "print records[0]['Entrezgene_type']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       AH004803                1143 bp    DNA     linear   BCT 26-AUG-2016\n",
      "DEFINITION  Escherichia coli fanF, fanG, and fanE genes, complete sequence.\n",
      "ACCESSION   AH004803 S79577 S79584\n",
      "VERSION     AH004803.2  GI:1060041767\n",
      "KEYWORDS    .\n",
      "SOURCE      Escherichia coli\n",
      "  ORGANISM  Escherichia coli\n",
      "            Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales;\n",
      "            Enterobacteriaceae; Escherichia.\n",
      "REFERENCE   1  (bases 1 to 1143)\n",
      "  AUTHORS   Lee,J.H. and Isaacson,R.E.\n",
      "  TITLE     Expression of the gene cluster associated with the Escherichia coli\n",
      "            pilus adhesin K99\n",
      "  JOURNAL   Infect. Immun. 63 (10), 4143-4149 (1995)\n",
      "   PUBMED   7558331\n",
      "COMMENT     On or before Aug 26, 2016 this sequence version replaced\n",
      "            gi:1168149, gi:1168150, gi:1683568.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..1143\n",
      "                     /organism=\"Escherichia coli\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /db_xref=\"taxon:562\"\n",
      "     misc_feature    1..1143\n",
      "                     /note=\"Original segmented set titles: K99 cluster:\n",
      "                     fanF...fanG {promoter} [Escherichia coli, Genomic, 3\n",
      "                     genes, 1043 nt 2 segments]; K99 cluster: fanF...fanG\n",
      "                     {promoter} [Escherichia coli, Genomic, 3 genes, 818 nt,\n",
      "                     segment 1 of 2]; K99 cluster: fanF...fanG {promoter}\n",
      "                     [Escherichia coli, Genomic, 3 genes, 225 nt, segment 2 of\n",
      "                     2]\"\n",
      "     gene            1..818\n",
      "                     /gene=\"fanF\"\n",
      "     mRNA            112..>818\n",
      "                     /gene=\"fanF\"\n",
      "     mRNA            340..>818\n",
      "                     /gene=\"fanF\"\n",
      "     mRNA            782..>818\n",
      "                     /gene=\"fanF\"\n",
      "     gene            801..818\n",
      "                     /gene=\"fanG\"\n",
      "     mRNA            956..>1043\n",
      "     gene            976..1043\n",
      "                     /gene=\"fanE\"\n",
      "     gap             1044..1143\n",
      "                     /estimated_length=unknown\n",
      "ORIGIN      \n",
      "        1 accagtcagg ctgggatggt aaacgttatg ctcaattaca tctttattca tcaggtgcct\n",
      "       61 tatgtgaaag tgtcagtgga gatgggatta catttaggtc aaatgtgtcc gggctgtcat\n",
      "      121 ggcgttttcc caatggcata ccataccact gtgcagcagg ccaaataaat cttggcggta\n",
      "      181 taaaatatgc ggatagaaat ggtaaagtta cctggaatcc tggtgaacta cgacatgaaa\n",
      "      241 tatttttaag agtggataac agatttgatt tcagtaaaag cagaacattt tctgtaaaca\n",
      "      301 caatttctgg tagaggagga ttaggtggag acagctcagt agttatacct ctcataggga\n",
      "      361 gttcatttaa ctattcctat tctaacatcg ctacctgcac tttgactggc ccaagtgaag\n",
      "      421 tgaatttcaa cactgtaacc acgtcagatg tactcaaagg aacaacacat cgtgatctta\n",
      "      481 acttaagggc agaatgtagg aacagggggg ctagcttagg actcaatttt aaatttgagc\n",
      "      541 ctcagtataa agatgtttct gcaaataaac agggcgtatt ttatgcaaaa aacaccagtg\n",
      "      601 ggaagcctta cttataaatt aacgaaaaaa gcagacgctt ctgctattcc actgaatgaa\n",
      "      661 tttgtcaaac ttattggttg aggataaagt taatatacat acagggaata ctattccatt\n",
      "      721 attataaccc ttcaaaaggg agacgggaaa atattgctac tggtaaaata gaaacattcc\n",
      "      781 tgaatgtcac aatggaacat atgtgatgaa aaaattatcg cagttaagaa tagcaatgga\n",
      "      841 gagattattt cttttgtcaa taacaatggg atttcacttc ttaatattga caagaatatg\n",
      "      901 agagagtctt ttttggatca tgtaccataa gtacttcggg acaaaagata atttatcaga\n",
      "      961 aattcaggag gtactgtgaa taaatttatt tctattatag ctctctgtgt attcagctct\n",
      "     1021 tacgctaatg ccgcttttac tctnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "     1081 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "     1141 nnn\n",
      "//\n",
      "\n",
      "LOCUS       NZ_LXFJ01000000      5263381 bp    DNA     linear   BCT 26-AUG-2016\n",
      "DEFINITION  Escherichia coli strain NA703, whole genome shotgun sequencing\n",
      "            project.\n",
      "ACCESSION   NZ_LXFJ00000000\n",
      "VERSION     NZ_LXFJ00000000.1  GI:1060037710\n",
      "DBLINK      BioProject: PRJNA224116\n",
      "            BioSample: SAMN04917426\n",
      "            Assembly: GCF_001713545.1\n",
      "KEYWORDS    WGS; RefSeq.\n",
      "SOURCE      Escherichia coli\n",
      "  ORGANISM  Escherichia coli\n",
      "            Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales;\n",
      "            Enterobacteriaceae; Escherichia.\n",
      "REFERENCE   1  (bases 1 to 5263381)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Molecular epidemiology of New Delhi metallo-beta-lactamase (NDM)\n",
      "            producing Extraintestinal Pathogenic E. coli (ExPEC) from India\n",
      "  JOURNAL   Unpublished\n",
      "REFERENCE   2  (bases 1 to 5263381)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (28-APR-2016) Biotechnology and Bioinformatics,\n",
      "            University of Hyderabad, Prof. C. R. Rao Road, Central University\n",
      "            P. O., Hyderabad, Telangana 500046, India\n",
      "COMMENT     REFSEQ INFORMATION: The reference sequence was derived from\n",
      "            LXFJ00000000.\n",
      "            The Escherichia coli whole genome shotgun (WGS) project has the\n",
      "            project accession NZ_LXFJ00000000.  This version of the project\n",
      "            (01) has the accession number NZ_LXFJ01000000, and consists of\n",
      "            sequences LXFJ01000001-LXFJ01000211.\n",
      "            Annotation was added by the NCBI Prokaryotic Genome Annotation\n",
      "            Pipeline (released 2013). Information about the Pipeline can be\n",
      "            found here: https://www.ncbi.nlm.nih.gov/genome/annotation_prok/\n",
      "            \n",
      "            ##Genome-Assembly-Data-START##\n",
      "            Assembly Method       :: SPAdes v. 3.6.1\n",
      "            Genome Coverage       :: 92.0x\n",
      "            Sequencing Technology :: Illumina MiSeq\n",
      "            ##Genome-Assembly-Data-END##\n",
      "            \n",
      "            ##Genome-Annotation-Data-START##\n",
      "            Annotation Provider               :: NCBI\n",
      "            Annotation Date                   :: 08/26/2016 14:21:13\n",
      "            Annotation Pipeline               :: NCBI Prokaryotic Genome\n",
      "                                                 Annotation Pipeline\n",
      "            Annotation Method                 :: Best-placed reference protein\n",
      "                                                 set; GeneMarkS+\n",
      "            Annotation Software revision      :: 3.3\n",
      "            Features Annotated                :: Gene; CDS; rRNA; tRNA; ncRNA;\n",
      "                                                 repeat_region\n",
      "            Genes (total)                     :: 5,357\n",
      "            CDS (total)                       :: 5,247\n",
      "            Genes (coding)                    :: 5,030\n",
      "            CDS (coding)                      :: 5,030\n",
      "            Genes (RNA)                       :: 110\n",
      "            rRNAs                             :: 8, 3, 5 (5S, 16S, 23S)\n",
      "            complete rRNAs                    :: 8 (5S)\n",
      "            partial rRNAs                     :: 3, 5 (16S, 23S)\n",
      "            tRNAs                             :: 79\n",
      "            ncRNAs                            :: 15\n",
      "            Pseudo Genes (total)              :: 217\n",
      "            Pseudo Genes (ambiguous residues) :: 0 of 217\n",
      "            Pseudo Genes (frameshifted)       :: 60 of 217\n",
      "            Pseudo Genes (incomplete)         :: 141 of 217\n",
      "            Pseudo Genes (internal stop)      :: 36 of 217\n",
      "            Pseudo Genes (multiple problems)  :: 20 of 217\n",
      "            CRISPR Arrays                     :: 2\n",
      "            ##Genome-Annotation-Data-END##\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..5263381\n",
      "                     /organism=\"Escherichia coli\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /strain=\"NA703\"\n",
      "                     /isolation_source=\"Pus\"\n",
      "                     /host=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:562\"\n",
      "                     /country=\"India: Pune\"\n",
      "                     /lat_lon=\"18.552185270984406 N 73.85156247764826 E\"\n",
      "                     /collection_date=\"19-Nov-2014\"\n",
      "                     /collected_by=\"University of Hyderabad\"\n",
      "WGS         LXFJ01000001-LXFJ01000211\n",
      "WGS_SCAFLD  NZ_LXFJ01000001-NZ_LXFJ01000211\n",
      "//\n",
      "\n",
      "LOCUS       NZ_LXFJ01000211          505 bp    DNA     linear   CON 26-AUG-2016\n",
      "DEFINITION  Escherichia coli strain NA703 contig211, whole genome shotgun\n",
      "            sequence.\n",
      "ACCESSION   NZ_LXFJ01000211 NZ_LXFJ00000000\n",
      "VERSION     NZ_LXFJ01000211.1  GI:1060037694\n",
      "DBLINK      BioProject: PRJNA224116\n",
      "            BioSample: SAMN04917426\n",
      "            Assembly: GCF_001713545.1\n",
      "KEYWORDS    WGS; RefSeq.\n",
      "SOURCE      Escherichia coli\n",
      "  ORGANISM  Escherichia coli\n",
      "            Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales;\n",
      "            Enterobacteriaceae; Escherichia.\n",
      "REFERENCE   1  (bases 1 to 505)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Molecular epidemiology of New Delhi metallo-beta-lactamase (NDM)\n",
      "            producing Extraintestinal Pathogenic E. coli (ExPEC) from India\n",
      "  JOURNAL   Unpublished\n",
      "REFERENCE   2  (bases 1 to 505)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (28-APR-2016) Biotechnology and Bioinformatics,\n",
      "            University of Hyderabad, Prof. C. R. Rao Road, Central University\n",
      "            P. O., Hyderabad, Telangana 500046, India\n",
      "COMMENT     REFSEQ INFORMATION: The reference sequence was derived from\n",
      "            LXFJ01000211.\n",
      "            Annotation was added by the NCBI Prokaryotic Genome Annotation\n",
      "            Pipeline (released 2013). Information about the Pipeline can be\n",
      "            found here: https://www.ncbi.nlm.nih.gov/genome/annotation_prok/\n",
      "            \n",
      "            ##Genome-Annotation-Data-START##\n",
      "            Annotation Provider               :: NCBI\n",
      "            Annotation Date                   :: 08/26/2016 14:21:13\n",
      "            Annotation Pipeline               :: NCBI Prokaryotic Genome\n",
      "                                                 Annotation Pipeline\n",
      "            Annotation Method                 :: Best-placed reference protein\n",
      "                                                 set; GeneMarkS+\n",
      "            Annotation Software revision      :: 3.3\n",
      "            Features Annotated                :: Gene; CDS; rRNA; tRNA; ncRNA;\n",
      "                                                 repeat_region\n",
      "            Genes (total)                     :: 5,357\n",
      "            CDS (total)                       :: 5,247\n",
      "            Genes (coding)                    :: 5,030\n",
      "            CDS (coding)                      :: 5,030\n",
      "            Genes (RNA)                       :: 110\n",
      "            rRNAs                             :: 8, 3, 5 (5S, 16S, 23S)\n",
      "            complete rRNAs                    :: 8 (5S)\n",
      "            partial rRNAs                     :: 3, 5 (16S, 23S)\n",
      "            tRNAs                             :: 79\n",
      "            ncRNAs                            :: 15\n",
      "            Pseudo Genes (total)              :: 217\n",
      "            Pseudo Genes (ambiguous residues) :: 0 of 217\n",
      "            Pseudo Genes (frameshifted)       :: 60 of 217\n",
      "            Pseudo Genes (incomplete)         :: 141 of 217\n",
      "            Pseudo Genes (internal stop)      :: 36 of 217\n",
      "            Pseudo Genes (multiple problems)  :: 20 of 217\n",
      "            CRISPR Arrays                     :: 2\n",
      "            ##Genome-Annotation-Data-END##\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..505\n",
      "                     /organism=\"Escherichia coli\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /strain=\"NA703\"\n",
      "                     /isolation_source=\"Pus\"\n",
      "                     /host=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:562\"\n",
      "                     /country=\"India: Pune\"\n",
      "                     /lat_lon=\"18.552185270984406 N 73.85156247764826 E\"\n",
      "                     /collection_date=\"19-Nov-2014\"\n",
      "                     /collected_by=\"University of Hyderabad\"\n",
      "     gene            <1..>505\n",
      "                     /locus_tag=\"A6803_RS26785\"\n",
      "                     /old_locus_tag=\"A6803_26795\"\n",
      "     CDS             <1..>505\n",
      "                     /locus_tag=\"A6803_RS26785\"\n",
      "                     /old_locus_tag=\"A6803_26795\"\n",
      "                     /inference=\"EXISTENCE: similar to AA\n",
      "                     sequence:RefSeq:WP_000782452.1\"\n",
      "                     /note=\"Derived by automated computational analysis using\n",
      "                     gene prediction method: Protein Homology.\"\n",
      "                     /codon_start=3\n",
      "                     /transl_table=11\n",
      "                     /product=\"conjugal transfer protein TraT\"\n",
      "                     /protein_id=\"WP_069067779.1\"\n",
      "                     /db_xref=\"GI:1060037266\"\n",
      "                     /translation=\"PASERTVFLQIKNTSDKDMSGLQGKIADAVKAKGYQVVTSPDAA\n",
      "                     YYWIQANVLKADKMDLRESQGWLNRGYEGAAVGAALGAGITGYNSNSAGATLGVGLAA\n",
      "                     GLVGMAADAMVEDVNYTMITDVQIAERTKATVTTDNVAALRQGTSGAKIQTSTETGNQ\n",
      "                     HKYQTRVV\"\n",
      "CONTIG      join(LXFJ01000211.1:1..505)\n",
      "//\n",
      "\n",
      "LOCUS       NZ_LXFJ01000210          511 bp    DNA     linear   CON 26-AUG-2016\n",
      "DEFINITION  Escherichia coli strain NA703 contig210, whole genome shotgun\n",
      "            sequence.\n",
      "ACCESSION   NZ_LXFJ01000210 NZ_LXFJ00000000\n",
      "VERSION     NZ_LXFJ01000210.1  GI:1060037692\n",
      "DBLINK      BioProject: PRJNA224116\n",
      "            BioSample: SAMN04917426\n",
      "            Assembly: GCF_001713545.1\n",
      "KEYWORDS    WGS; RefSeq.\n",
      "SOURCE      Escherichia coli\n",
      "  ORGANISM  Escherichia coli\n",
      "            Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales;\n",
      "            Enterobacteriaceae; Escherichia.\n",
      "REFERENCE   1  (bases 1 to 511)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Molecular epidemiology of New Delhi metallo-beta-lactamase (NDM)\n",
      "            producing Extraintestinal Pathogenic E. coli (ExPEC) from India\n",
      "  JOURNAL   Unpublished\n",
      "REFERENCE   2  (bases 1 to 511)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (28-APR-2016) Biotechnology and Bioinformatics,\n",
      "            University of Hyderabad, Prof. C. R. Rao Road, Central University\n",
      "            P. O., Hyderabad, Telangana 500046, India\n",
      "COMMENT     REFSEQ INFORMATION: The reference sequence was derived from\n",
      "            LXFJ01000210.\n",
      "            Annotation was added by the NCBI Prokaryotic Genome Annotation\n",
      "            Pipeline (released 2013). Information about the Pipeline can be\n",
      "            found here: https://www.ncbi.nlm.nih.gov/genome/annotation_prok/\n",
      "            \n",
      "            ##Genome-Annotation-Data-START##\n",
      "            Annotation Provider               :: NCBI\n",
      "            Annotation Date                   :: 08/26/2016 14:21:13\n",
      "            Annotation Pipeline               :: NCBI Prokaryotic Genome\n",
      "                                                 Annotation Pipeline\n",
      "            Annotation Method                 :: Best-placed reference protein\n",
      "                                                 set; GeneMarkS+\n",
      "            Annotation Software revision      :: 3.3\n",
      "            Features Annotated                :: Gene; CDS; rRNA; tRNA; ncRNA;\n",
      "                                                 repeat_region\n",
      "            Genes (total)                     :: 5,357\n",
      "            CDS (total)                       :: 5,247\n",
      "            Genes (coding)                    :: 5,030\n",
      "            CDS (coding)                      :: 5,030\n",
      "            Genes (RNA)                       :: 110\n",
      "            rRNAs                             :: 8, 3, 5 (5S, 16S, 23S)\n",
      "            complete rRNAs                    :: 8 (5S)\n",
      "            partial rRNAs                     :: 3, 5 (16S, 23S)\n",
      "            tRNAs                             :: 79\n",
      "            ncRNAs                            :: 15\n",
      "            Pseudo Genes (total)              :: 217\n",
      "            Pseudo Genes (ambiguous residues) :: 0 of 217\n",
      "            Pseudo Genes (frameshifted)       :: 60 of 217\n",
      "            Pseudo Genes (incomplete)         :: 141 of 217\n",
      "            Pseudo Genes (internal stop)      :: 36 of 217\n",
      "            Pseudo Genes (multiple problems)  :: 20 of 217\n",
      "            CRISPR Arrays                     :: 2\n",
      "            ##Genome-Annotation-Data-END##\n",
      "            \n",
      "            ##Genome-Assembly-Data-START##\n",
      "            Assembly Method       :: SPAdes v. 3.6.1\n",
      "            Genome Coverage       :: 92.0x\n",
      "            Sequencing Technology :: Illumina MiSeq\n",
      "            ##Genome-Assembly-Data-END##\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..511\n",
      "                     /organism=\"Escherichia coli\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /strain=\"NA703\"\n",
      "                     /isolation_source=\"Pus\"\n",
      "                     /host=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:562\"\n",
      "                     /country=\"India: Pune\"\n",
      "                     /lat_lon=\"18.552185270984406 N 73.85156247764826 E\"\n",
      "                     /collection_date=\"19-Nov-2014\"\n",
      "                     /collected_by=\"University of Hyderabad\"\n",
      "     gene            <1..>511\n",
      "                     /locus_tag=\"A6803_RS26780\"\n",
      "                     /old_locus_tag=\"A6803_26790\"\n",
      "                     /pseudo\n",
      "     CDS             <1..>511\n",
      "                     /locus_tag=\"A6803_RS26780\"\n",
      "                     /old_locus_tag=\"A6803_26790\"\n",
      "                     /inference=\"COORDINATES: ab initio prediction:GeneMarkS+\"\n",
      "                     /note=\"incomplete; too short partial abutting assembly\n",
      "                     gap; missing start and stop; Derived by automated\n",
      "                     computational analysis using gene prediction method:\n",
      "                     GeneMarkS+.\"\n",
      "                     /pseudo\n",
      "                     /codon_start=3\n",
      "                     /transl_table=11\n",
      "                     /product=\"hypothetical protein\"\n",
      "CONTIG      join(LXFJ01000210.1:1..511)\n",
      "//\n",
      "\n",
      "LOCUS       NZ_LXFJ01000209          530 bp    DNA     linear   CON 26-AUG-2016\n",
      "DEFINITION  Escherichia coli strain NA703 contig209, whole genome shotgun\n",
      "            sequence.\n",
      "ACCESSION   NZ_LXFJ01000209 NZ_LXFJ00000000\n",
      "VERSION     NZ_LXFJ01000209.1  GI:1060037691\n",
      "DBLINK      BioProject: PRJNA224116\n",
      "            BioSample: SAMN04917426\n",
      "            Assembly: GCF_001713545.1\n",
      "KEYWORDS    WGS; RefSeq.\n",
      "SOURCE      Escherichia coli\n",
      "  ORGANISM  Escherichia coli\n",
      "            Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacteriales;\n",
      "            Enterobacteriaceae; Escherichia.\n",
      "REFERENCE   1  (bases 1 to 530)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Molecular epidemiology of New Delhi metallo-beta-lactamase (NDM)\n",
      "            producing Extraintestinal Pathogenic E. coli (ExPEC) from India\n",
      "  JOURNAL   Unpublished\n",
      "REFERENCE   2  (bases 1 to 530)\n",
      "  AUTHORS   Ranjan,A., Sabiha,S., Mondal,A., Hussain,A., Nandanwar,N.,\n",
      "            Kumar,N., Semmler,T., Jadhav,S., Tiwari,S.K., Wieler,L.H. and\n",
      "            Ahmed,N.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (28-APR-2016) Biotechnology and Bioinformatics,\n",
      "            University of Hyderabad, Prof. C. R. Rao Road, Central University\n",
      "            P. O., Hyderabad, Telangana 500046, India\n",
      "COMMENT     REFSEQ INFORMATION: The reference sequence was derived from\n",
      "            LXFJ01000209.\n",
      "            Annotation was added by the NCBI Prokaryotic Genome Annotation\n",
      "            Pipeline (released 2013). Information about the Pipeline can be\n",
      "            found here: https://www.ncbi.nlm.nih.gov/genome/annotation_prok/\n",
      "            \n",
      "            ##Genome-Annotation-Data-START##\n",
      "            Annotation Provider               :: NCBI\n",
      "            Annotation Date                   :: 08/26/2016 14:21:13\n",
      "            Annotation Pipeline               :: NCBI Prokaryotic Genome\n",
      "                                                 Annotation Pipeline\n",
      "            Annotation Method                 :: Best-placed reference protein\n",
      "                                                 set; GeneMarkS+\n",
      "            Annotation Software revision      :: 3.3\n",
      "            Features Annotated                :: Gene; CDS; rRNA; tRNA; ncRNA;\n",
      "                                                 repeat_region\n",
      "            Genes (total)                     :: 5,357\n",
      "            CDS (total)                       :: 5,247\n",
      "            Genes (coding)                    :: 5,030\n",
      "            CDS (coding)                      :: 5,030\n",
      "            Genes (RNA)                       :: 110\n",
      "            rRNAs                             :: 8, 3, 5 (5S, 16S, 23S)\n",
      "            complete rRNAs                    :: 8 (5S)\n",
      "            partial rRNAs                     :: 3, 5 (16S, 23S)\n",
      "            tRNAs                             :: 79\n",
      "            ncRNAs                            :: 15\n",
      "            Pseudo Genes (total)              :: 217\n",
      "            Pseudo Genes (ambiguous residues) :: 0 of 217\n",
      "            Pseudo Genes (frameshifted)       :: 60 of 217\n",
      "            Pseudo Genes (incomplete)         :: 141 of 217\n",
      "            Pseudo Genes (internal stop)      :: 36 of 217\n",
      "            Pseudo Genes (multiple problems)  :: 20 of 217\n",
      "            CRISPR Arrays                     :: 2\n",
      "            ##Genome-Annotation-Data-END##\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..530\n",
      "                     /organism=\"Escherichia coli\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /strain=\"NA703\"\n",
      "                     /isolation_source=\"Pus\"\n",
      "                     /host=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:562\"\n",
      "                     /country=\"India: Pune\"\n",
      "                     /lat_lon=\"18.552185270984406 N 73.85156247764826 E\"\n",
      "                     /collection_date=\"19-Nov-2014\"\n",
      "                     /collected_by=\"University of Hyderabad\"\n",
      "     gene            <1..245\n",
      "                     /locus_tag=\"A6803_RS26765\"\n",
      "                     /old_locus_tag=\"A6803_26775\"\n",
      "                     /pseudo\n",
      "     CDS             <1..245\n",
      "                     /locus_tag=\"A6803_RS26765\"\n",
      "                     /old_locus_tag=\"A6803_26775\"\n",
      "                     /inference=\"EXISTENCE: similar to AA\n",
      "                     sequence:RefSeq:WP_001230282.1\"\n",
      "                     /note=\"incomplete; too short partial abutting assembly\n",
      "                     gap; missing start; Derived by automated computational\n",
      "                     analysis using gene prediction method: Protein Homology.\"\n",
      "                     /pseudo\n",
      "                     /codon_start=3\n",
      "                     /transl_table=11\n",
      "                     /product=\"enterobacterial Ail/Lom family protein\"\n",
      "     gene            complement(246..419)\n",
      "                     /locus_tag=\"A6803_RS26770\"\n",
      "                     /old_locus_tag=\"A6803_26780\"\n",
      "     CDS             complement(246..419)\n",
      "                     /locus_tag=\"A6803_RS26770\"\n",
      "                     /old_locus_tag=\"A6803_26780\"\n",
      "                     /inference=\"EXISTENCE: similar to AA\n",
      "                     sequence:RefSeq:WP_000495484.1\"\n",
      "                     /note=\"Derived by automated computational analysis using\n",
      "                     gene prediction method: Protein Homology.\"\n",
      "                     /codon_start=1\n",
      "                     /transl_table=11\n",
      "                     /product=\"hypothetical protein\"\n",
      "                     /protein_id=\"WP_001382782.1\"\n",
      "                     /db_xref=\"GI:485757274\"\n",
      "                     /translation=\"MFTTTAVLLRFAFSWIVQFCTGFPVPSFSTPEIFTAIFTPQKSP\n",
      "                     PEPAGCHNTVLPG\"\n",
      "     gene            412..>530\n",
      "                     /locus_tag=\"A6803_RS26775\"\n",
      "                     /old_locus_tag=\"A6803_26785\"\n",
      "                     /pseudo\n",
      "     CDS             412..>530\n",
      "                     /locus_tag=\"A6803_RS26775\"\n",
      "                     /old_locus_tag=\"A6803_26785\"\n",
      "                     /inference=\"COORDINATES: ab initio prediction:GeneMarkS+\"\n",
      "                     /note=\"incomplete; too short partial abutting assembly\n",
      "                     gap; missing stop; Derived by automated computational\n",
      "                     analysis using gene prediction method: GeneMarkS+.\"\n",
      "                     /pseudo\n",
      "                     /codon_start=1\n",
      "                     /transl_table=11\n",
      "                     /product=\"phage tail protein\"\n",
      "CONTIG      join(LXFJ01000209.1:1..530)\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "record = Entrez.read(Entrez.esearch(db = 'nuccore', term = 'e coli[Orgn]'))\n",
    "ids = ','.join(record['IdList'][:5])\n",
    "handle = Entrez.efetch(db = 'nuccore', id = ids, rettype = 'gb', retmode = 'text')\n",
    "print handle.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle.read()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
